src/components/PhoneCards/PhonesForHotPrices.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useContext } from 'react';
+import { DispatchContext, StateContext } from '../../Store';
+import { Product } from '../../types';
+import { Link } from 'react-router-dom';
+import stroke from '../../assets/img/icons/Vector (Stroke).svg';
+import { useTheme } from '../../contexts/ThemeContext';
+
+type Props = {
+  image: string;
+  name: string;
+  fullPrice: number;
+  price: number;
+  screen: string;
+  capacity: string;
+  ram: string;
+  product: Product;
+};
+
+export const PhonesForHotPrices: React.FC<Props> = ({
+  image,
+  name,
+  fullPrice,
+  price,
+  screen,
+  capacity,
+  ram,
+  product,
+}) => {
+  const dispatch = useContext(DispatchContext);
+  const state = useContext(StateContext);
+  const { favorites, basket } = state;
+  const { theme } = useTheme();
+
+  const isFavorite = !!favorites.find(item => item === product.itemId);
+  const added = basket.find(
+    (item: { itemId: string }) => item.itemId === product.itemId,
+  );
+
+  const addToFavorites = () => {
+    if (!isFavorite) {
+      dispatch({
+        type: 'addToFavorites',
+        payload: product,
+      });
+    }
+
+    if (isFavorite) {
+      dispatch({
+        type: 'removeFromFavorites',
+        payload: { itemId: product.itemId },
+      });
+    }
+  };
+
+  const addToBascket = () => {
+    if (!added) {
+      dispatch({
+        type: 'addToBasket',
+        payload: product,
+      });
+    }
+
+    if (added) {
+      dispatch({
+        type: 'removeFromBasket',
+        payload: { itemId: product.itemId },
+      });
+    }
+  };
+
+  return theme === 'light' ? (
```

Comment:
better not to use conditional render, you can use hook useTheme() inside your card